### PR TITLE
[Cisco Umbrella] adding support for Cisco Managed S3 buckets and non SQS

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

Adds support for Non SQS type S3 buckets, including Cisco Managed S3 buckets.
Updates pipeline to support the newest proxy logs.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by P1llus** [comment]
I wish there was a better way to separate SQS and non SQS without being confusing about the settings :(

@kaiyan-sheng Tagging you as optional in case you also wanted to take a look, as it is S3 related.

**Review by efd6 — APPROVED** [review]

File: packages/cisco_umbrella/data_stream/log/manifest.yml [hunk 1]
```diff
@@ -11,9 +11,40 @@ streams:
...
+        description: >-
+          Required for Cisco Managed S3. When using S3 without SQS, a regex pattern to fit the folders is required. When using Cisco Managed S3, this should be the same value as Bucket List Prefix ending with a forward slash. Example: "1235_654vcasd23431e5dd6f7fsad457sdf1fd5/"
+      - name: number_of_workers
+        type: text
+        title: Number of Workers
+        multi: false
+        required: false
         show_user: true
-        description: URL of the AWS SQS queue that messages will be received from.
+        description: Required for Cisco Managed S3. Number of workers that will process the S3 objects listed. Default is 1.
```
Comment (aspacca) [inline packages/cisco_umbrella/data_stream/log/manifest.yml:47]: please, beware that this config param has no default (ie: default 0), if not set the underlying filebeat instance will exit. `Default is 1.` message might be misleading in this sense, not sure if it needs a better wording

**Comment by aspacca** [comment]
@P1llus 
minor: https://github.com/elastic/beats/blob/master/x-pack/filebeat/input/awss3/config.go#L32 `bucket_list_interval` config param is missing, not sure if you want to add.

also having the s3 direct polling as input in the agent might be tricky, see: https://www.elastic.co/guide/en/beats/filebeat/7.15/filebeat-input-aws-s3.html#_parallel_processing

```
When using the polling list of S3 bucket objects method be aware that if running multiple Filebeat instances, they can list the same S3 bucket at the same time. Since the state of the ingested S3 objects is persisted (upon processing a single list operation) in the path.data configuration and multiple Filebeat cannot share the same path.data this will produce repeated ingestion of the S3 object. Therefore, when using the polling list of S3 bucket objects method, scaling should be vertical, with a single bigger Filebeat instance and higher number_of_workers config value.
```

Since with integrations the common case is to have the same policy dispatched to multiple agents will break due to the above behaviour.
I think we should add some big "warning" about creating a dedicated policy for the integration when dealing with Cisco Managed S3 buckets and linking it to a single agent.

File: packages/cisco_umbrella/data_stream/log/manifest.yml (hunk 1 above)
Comment (aspacca) [reply packages/cisco_umbrella/data_stream/log/manifest.yml]: missed https://github.com/elastic/integrations/blob/57a49493c3c3f0c89bbcda46ae4e879b65a4a08a/packages/cisco_umbrella/data_stream/log/manifest.yml#L47

**Comment by P1llus** [comment]
Still adding some small fixes to the package

**Comment by P1llus** [comment]
Had to add a few changes to make everything work as expected:

If you use non SQS, we want to ignore the auditlogs, as they are not in CSV format, or any recognized format:
```
{{#if bucket_list_prefix}}
file_selectors: 
  - regex: {{bucket_list_prefix}}/dnslogs/.+
  - regex: {{bucket_list_prefix}}/proxylogs/.+
  - regex: {{bucket_list_prefix}}/cloudfirewalllogs/.+
  - regex: {{bucket_list_prefix}}/iplogs/.+
{{/if}}
```
Moved away from the GROK pattern, and to the CSV format that the original pipeline used to make it more correct.
Added support for the new Proxy logs.

**Comment by P1llus** [comment]
Package has been tested with a Cisco Managed S3 bucket and works correctly.

**Review by aspacca — APPROVED** [review]

File: packages/cisco_umbrella/_dev/build/docs/README.md [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
-datasets for receiving logs from an AWS S3 bucket using an SQS notification queue:
+datasets for receiving logs from an AWS S3 bucket using an SQS notification queue and Cisco Managed S3 bucket without SQS:
 
 - `log` dataset: supports Cisco Umbrella logs.
 
 ## Logs
 
 ### Umbrella
 
+When using Cisco Managed S3 buckets that does not use SQS there is no load balancing possibilities for multiple agents, a single agent should be configured to poll the S3 bucket for new and updated files, and the number of workers can be configured to scale vertically
```
Comment (kaiyan-sheng) [inline packages/cisco_umbrella/_dev/build/docs/README.md:12]:
```suggestion
When using Cisco Managed S3 buckets that does not use SQS there is no load balancing possibilities for multiple agents, a single agent should be configured to poll the S3 bucket for new and updated files, and the number of workers can be configured to scale vertically.
```

**Pull request merged**